examples/http3-server.rs

```diff
@@ -47,6 +47,7 @@ Options:
   --name <str>             Name of the server [default: quic.tech]
   --max-data BYTES         Connection-wide flow control limit [default: 10000000].
   --max-stream-data BYTES  Per-stream flow control limit [default: 1000000].
+  --max-streams STREAMS    Number of allowed concurrent streams [default: 100].
```

Comment:
I think we might want to expose independent concurrency values for uni and bidi streams. The comment applies to server.rs too.

With the same concurrency, In the best case, a user is just going to create an application that opens themselves up for receiving more junk peer-initiated unidirectional streams. In the worst case, max-streams (<= 2) the application will never give the peer enough stream credit for a successful H3 connection. They could sleepwalk into this situation when trying to play around with different levels of request concurrency - it's reasonable to want to control how many requests are allowed.

Reply:
Ok, done.